src/opera_disp_tms/search.py

```diff
@@ -56,6 +58,25 @@ def from_umm(cls, umm: dict) -> 'Granule':
             creation_date=creation_date,
         )
 
+    def __eq__(self, other) -> bool:
+        """
+        Compare two granules to see if they are represent the same data
+        Args:
+            other: Granule object to compare
+
+        Returns:
+            bool: True if they are identical, False otherwise
+        """
+        return (
+            self.frame_id == other.frame_id
+            and self.reference_date == other.reference_date
+            and self.secondary_date == other.secondary_date
+            and self.creation_date == other.creation_date
+        )
```

Comment:
I think this could be reduced to `return hash(self) == hash(other)`?

Reply:
This is what stack overflow recommends https://stackoverflow.com/questions/45164691/recommended-way-to-implement-eq-and-hash. I think it would fail if object isn't hashable